### PR TITLE
Add support for custom help templates.

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by jszwedko** [comment]
Hi @harshavardhana!

I really like this addition, but would you mind adding a test or two for it?

**Comment by harshavardhana** [comment]
> I really like this addition, but would you mind adding a test or two for it?

Sure i can do that @jszwedko

**Comment by harshavardhana** [comment]
Added tests as requested. @jszwedko

File: context.go [hunk 1]
```diff
@@ -186,9 +186,23 @@ func (a Args) First() string {
...
+	if len(a) == 1 {
+		return a
+	}
+	return []string(a)[:len(a)-1]
+}
+
 // Tail returns the rest of the arguments (not the first one)
 // or else an empty string slice
-func (a Args) Tail() []string {
+func (a Args) Tail() Args {
```
Comment (jszwedko) [inline context.go]: Did you mean to include this here? It feels orthogonal to the rest of the PR and breaks compatibility.

**Comment by harshavardhana** [comment]
> Overall this looks good @harshavardhana ; apologies for the delay! Just left one comment about changes that appear to be unrelated to the rest of this PR.

Thanks for the review.. @jszwedko updated. Removed the change.

File: help.go [hunk 2]
```diff
@@ -148,7 +179,11 @@ func ShowCommandHelp(ctx *Context, command string) error {
 
 	for _, c := range ctx.App.Commands {
 		if c.HasName(command) {
-			HelpPrinter(ctx.App.Writer, CommandHelpTemplate, c)
+			if c.CustomHelpTemplate != "" {
+				HelpPrinter(ctx.App.Writer, c.CustomHelpTemplate, c)
```
Comment (jszwedko) [inline help.go]: Should this be calling `HelpPrinterCustom`?

Comment (harshavardhana) [inline help.go]: Sure.. doesn't matter really.

**Comment by harshavardhana** [comment]
Updated @jszwedko

**Comment by harshavardhana** [comment]
@jszwedko anything pending here?

**Comment by harshavardhana** [comment]
@jszwedko anything else is needed ?

**Comment by harshavardhana** [comment]
@jszwedko can you tell me what is else needed to get this merged?

**Comment by jszwedko** [comment]
:+1: this looks good. Apologies for the severe delay in review; was on vacation for the past month and didn't keep up with my email as much as I thought I might.

**Comment by harshavardhana** [comment]
No problem @jszwedko
